### ISSUE TITLE
Cannot find module when installed globally

### ISSUE THREAD

**Issue by unional** [npm/npm#10422]
Some modules cannot be found when installed globally.

Say I have:
````
C:\Users\me\AppData\Roaming\npm
├── casperjs@1.1.0-beta3
├── gulp@3.9.0
├── jspm@0.16.14
├── jspm-p4@0.2.3
├── karma@0.13.15
├── npm@3.4.0
├── npm-check-updates@2.4.2
├── phantomjs@1.9.18
└── tsd@0.6.5
````

When I am at a folder that does not have local modules (e.g. `C:\Users\me`):
````
C:\Users\me> node
> require('jspm')
-> jspm;
> require('karma');
-> karma;
> require('jspm-p4');
-> Cannot find module (Error)
> require('npm-check-updates')
-> Cannot find module (Error)
````

Adding `NODE_PATH` explicitly pointing to `C:\Users\me\AppData\Roaming\npm\npm_modules` solves the problem, but that would consider as a workaround instead of a solution as some modules are working.

https://github.com/jspm/jspm-cli/issues/1284

`jspm-p4` is a module that I created and you can find it [here](https://github.com/unional/jspm-p4)

**Comment by othiym23** [comment]
I don't know your exact use case, but what you're describing is how [Node.js's module resolution algorithm](https://nodejs.org/api/modules.html#modules_all_together) works – npm has nothing to do with how modules are resolved by Node. Node, unlike many dynamic languages, doesn't put global modules on a lookup path by default. The predominant usage pattern is locally installed modules. (In fact, there have been several, thus far unsuccessful, efforts to deprecate and remove `NODE_MODULES`.)

The mystery, to me, is that `require('jspm')` works in `C:\Users\me` – it shouldn't, unless you've set `NODE_PATH` beforehand. If you dig into that a little bit, you might figure out what's going on and what you need to do to make things work. I also don't know how jspm wants you to write extensions, but I'm pretty sure you're going to need to do something more than just globally installing your plugin.

Either way, these issues are outside the scope of npm – it's just the delivery agent here. Good luck getting this figured out!

**Comment by unional** [comment]
Ok, I'll file the issue on node instead.

Yeah, don't know why some works and some not. That's what bugging me. :)

**Comment by Sheeryada** [comment]
@unional appreciate it.. ! I have the similar issue as well...

**Comment by unional** [comment]
Closed the node issue. It is by design that it should not work.  Don't know why it works for some modules O.o

**Comment by huan** [comment]
As a 2 years node developer, it's a surprise for me that node will not search global installed modules for `require`.

I ran into this issue today, and after search a lot, I found that many people have the same question like me: why I got an error like "Error: Cannot find module 'npm'" after installed the module globally? 

Then I start to think about if there's a problem with the name **global** and the parameter **--global**

As the document said:

```
· npm install:
           In global mode (ie, with -g or --global appended to the command),
           it installs the current package context (ie, the current working
           directory) as a global package.
```

It makes the reader consider that:
1. Local install is for the current directory project, and 
1. Global install should be used for all the system projects.

But in fact, the "global install" is just for "command line tools".

My suggestion will be: 
1. deprecate the `--global` name
1. rename `--global` to `--cli`

That would be better for understanding and will save lots of time for the developers like me and others who asked the same question.

@othiym23
